src/vivarium/framework/results/interface.py

```diff
@@ -182,7 +180,7 @@ def register_stratified_observation(
         additional_stratifications: list[str] = [],
         excluded_stratifications: list[str] = [],
         aggregator_sources: list[str] | None = None,
-        aggregator: Callable[[pd.DataFrame], float | pd.Series] = len,
+        aggregator: Callable[[pd.DataFrame], float | pd.Series[Any]] = len,
```

Comment:
Arent the series' always number likes?

Reply:
I think they are in practice, but in principle they don't need to be? Here's an example:

We're recording blood pressure for a population and we're stratifying by age group. We don't actually care about the actual mean blood pressure - we want to categorize population-level mean blood pressure into low, medium, and high. Then the aggregator function would return a string, not a number.

Reply:
In that case I'll leave it as Any. Does that mean we want to update the scalar type as well?